modules/api-support-deprecation-policy.adoc

```diff
@@ -0,0 +1,103 @@
+// Module included in the following assemblies:
+//
+// * rest_api/understanding-api-support-tiers.adoc
+
+[id="api-deprecation-policy_{context}"]
+= API deprecation policy
+
+{product-title} is composed of many components sourced from many upstream communities. It is anticipated that the set of components, the associated API interfaces, and correlated features will evolve over time and might require formal deprecation in order to remove the capability.
+
+[id="deprecating-parts-of-the-api_{context}"]
+== Deprecating parts of the API
+
+{product-title} is a distributed system where multiple components interact with a shared state managed by the cluster control plane through a set of structured APIs. Per Kubernetes conventions, each API presented by {product-title} is associated with a group identifier and each API group is independently versioned.  Each API group is managed in a distinct upstream community including Kubernetes, Metal3, Multus, Operator Framework, Open Cluster Management, OpenShift itself, and more.
+
+While each upstream community might define their own unique deprecation policy for a given API group and version, Red Hat normalizes the community specific policy to one of the compatibility levels defined prior based on our integration in and awareness of each upstream community to simplify end-user consumption and support.
+
+The deprecation policy and schedule for APIs vary by compatibility level.
+
+The deprecation policy covers all elements of the API including:
+
+* REST resources, also known as API objects
+* Fields of REST resources
+* Annotations on REST resources, excluding version-specific qualifiers
+* Enumerated or constant values
+
+Other than the most recent API version in each group, older API versions must be supported after their announced deprecation for a duration of no less than:
+
+[cols="2",options="header"]
+|===
+|API tier
+|Duration
+
+|Tier 1
+|12 months or 3 releases from the announcement of deprecation, whichever is longer.
+
+|Tier 2
+|9 months or 3 releases from the announcement of deprecation, whichever is longer.
+
+|Tier 3
+|See the component-specific schedule.
+
+|Tier 4
+|None. No compatibility is guaranteed.
+
+|===
+
+The following rules apply to all tier 1 APIs:
+
+* API elements can only be removed by incrementing the version of the group.
+* API objects must be able to round-trip between API versions without information loss, with the exception of whole REST resources that do not exist in some versions.
```

Comment:
these two bullets contradict each other.  if you remove an api element(field) in v2 of an apigroup, then you can't possibly roundtrip to v1 without information loss.

Reply:
per derek we do this by storing the data from the missing fields as annotations on the older api object.